### PR TITLE
Added data types for the wakefield keys

### LINKED ISSUE #4347: Random samples and Wakefield package
This seems a very nice package - don't know why it is called Wakefield - for the use of R-Instat in teaching.  I suggest:
a) We add it to the packages that are installed with R-Instat.  Then it can already be used in our calculator to generate interesting random samples.
b) We consider adding it in 3 places, namely:
1) In the **File > New** dialogue.
2) In the **Prepare > Generate > Random Samples** dialogue - where I suggest it should be a sub-dialogue - perhaps called Options or even Wakefield. 
3) Possibly also a keyboard in the calculator - that could be fun!  There it would generate a single sample (column) whereas the dialogues could allow the generation of multiple columns.

What I am not clear is on the structure of the sub-dialogue.  Are there any principles we could use here?

I would like this perhaps to be considered by our team in AMI (the documentation team) in discussion with David.

I suggest that, in the short term, it could become a great new keyboard in our calculator and this would be relatively simple and hence a possible job for a newcomer, perhaps under supervision from Shadrack?  One complication here is that the package suggests a sensible name for the resulting column.  We could ignore this feature or try to incorporate it.

This is also an example of a package where we could investigate how it can be used in teaching?  I like some of the functions for different types of data and also the way it suggests an automatic type fore the resulting column.  So it generates factors and ordered factors and logical columns, etc.

In the File > New and Random Samples dialogue I suggest the Climatic > Climdex sub-dialogues give a possible structure.  But I am not sure how to allow for the probabilities.  That isn't yet so easy with our dialogue structure.

### PR DESCRIPTION
**Pull request by Wycklife**
Fixes #4347
 This fixes the last part of the issue #4347
@africanmathsinitiative/developers This can be reviewed.

### CODE REVIEW

**Comment by Wycklife** [comment]
@rdstern I have added the data types for the keys proposed, Area and Zip keys are missing, should I add them? Otherwise, kindly have a look at the data types.

File: instat/ucrCalculator.vb [hunk 1]
```diff
@@ -1695,6 +1720,8 @@ Public Class ucrCalculator
     End Sub
 
     Private Sub cmdSex_Inclusive_Click(sender As Object, e As EventArgs) Handles cmdSex_Inclusive.Click
+        ucrReceiverForCalculation.SetDataType("factor")
+        ucrReceiverForCalculation.SetDataType("factors")
         If chkShowArguments.Checked Then
```
Comment (maxwellfundi) [inline instat/ucrCalculator.vb]: @Wycklife  why do you have these two types here - `factor `and `factors`?

Comment (Wycklife) [reply instat/ucrCalculator.vb]: @maxwellfundi Thanks for that observation. Am correcting this now.

**Comment by rdstern** [comment]
@Wycklife and @maxwellfundi I must have done something wrong here.  I thought I took your branch - it is with the improved keyboard.  I expected to see for example coin replaced by coin (f).  And also Hair replaced by hair (f), i.e. also the lower case h.  But it seems to be the same as before.

File: instat/ucrCalculator.vb [hunk 2]
```diff
@@ -1599,6 +1615,8 @@ Public Class ucrCalculator
     End Sub
 
     Private Sub cmdLorem_Ipsum_Click(sender As Object, e As EventArgs) Handles cmdLorem_ipsum.Click
+        ucrReceiverForCalculation.SetDataType("character")
+        ucrReceiverForCalculation.SetDataType("character")
```
Comment (maxwellfundi) [inline instat/ucrCalculator.vb]: @Wycklife  there seeems to have some repeated code here. Please get rid of one of these lines.

File: instat/ucrCalculator.vb [hunk 3]
```diff
@@ -1464,6 +1472,7 @@ Public Class ucrCalculator
     End Sub
 
     Private Sub cmdEducation_Click(sender As Object, e As EventArgs) Handles cmdEducation.Click
+        ucrReceiverForCalculation.SetDataType("factors")
```
Comment (maxwellfundi) [inline instat/ucrCalculator.vb]: @Wycklife here you also have `factors` other than `factor`

File: instat/ucrCalculator.vb [hunk 4]
```diff
@@ -1360,6 +1361,7 @@ Public Class ucrCalculator
     End Sub
 
     Private Sub cmdAnswer_Click(sender As Object, e As EventArgs) Handles cmdAnswer.Click
+        ucrReceiverForCalculation.SetDataType("factors")
```
Comment (maxwellfundi) [inline instat/ucrCalculator.vb]: @Wycklife `factors `here again- change it to `factor`

**Comment by rdstern** [comment]
@maxwellfundi and @Wycklife I still wonder if there is a misunderstanding on what is needed here.  I think the functions automatically specify the type of the resulting variables.  What I was asking for was that the visible keys on the keyboard carry this information.  So when you see the key, then you know what type of variable to expect so the key itself says hair (f).  I got the variable types by simply generating "the works" and seeing what types of variable results.

**Comment by Wycklife** [comment]
@rdstern There was a misunderstanding on this, let me redo the issue. Should I close this pull request?

**Comment by rdstern** [comment]
Ok,  sorry I didn't detect earlier.  Yes, I think it could be a new pull request.

**Comment by Wycklife** [comment]
I'm closing this due to the unwanted features

**Pull request closed without merging**